src/auth_client_demo_assets/vue/store/auth.js

```diff
@@ -0,0 +1,88 @@
+import { defineStore } from "pinia";
+import { AuthClient } from "@dfinity/auth-client";
+import { createActor, canisterId } from "../../../declarations/whoami";
+
+const defaultOptions = {
+  /**
+   *  @type {import("@dfinity/auth-client").AuthClientCreateOptions}
+   */
+  createOptions: {
+    idleOptions: {
+      // Set to true if you do not want idle functionality
+      disableIdle: true,
+    },
+  },
+  /**
+   * @type {import("@dfinity/auth-client").AuthClientLoginOptions}
+   */
+  loginOptions: {
+    identityProvider:
+      process.env.DFX_NETWORK === "ic"
+        ? "https://identity.ic0.app/#authorize"
+        : `http://localhost:4943?canisterId=${process.env.CANISTER_ID_INTERNET_IDENTITY}#authorize`,
+  },
+};
+
+function actorFromIdentity(identity) {
+  return createActor(canisterId, {
+    agentOptions: {
+      identity,
+    },
+  });
+}
+
+export const useAuthStore = defineStore("auth", {
+  id: "auth",
+  state: () => {
+    return {
+      isAuthenticated: null,
+      authClient: null,
+      identity: null,
+      whoamiActor: null,
+    };
+  },
+  actions: {
+    async init() {
+      const authClient = await AuthClient.create(defaultOptions.createOptions);
+      this.authClient = authClient;
+      const isAuthenticated = await authClient.isAuthenticated();
+      const identity = isAuthenticated ? authClient.getIdentity() : null;
+      const whoamiActor = identity ? actorFromIdentity(identity) : null;
+
+      this.isAuthenticated = isAuthenticated;
+      this.identity = identity;
+      this.whoamiActor = whoamiActor;
+    },
+    async login() {
+      /*
+       * Note: for Vue, a proxy of the authClient cannot be used for "login"
+       * because the `postMessage` API is interfered with. We can re-create the client for the login, and then set it after the login is complete.
+       * All other auth-client methods and attributes can use the proxy.
+       */
+      const authClient = await AuthClient.create(defaultOptions.createOptions);
+      authClient.login({
```

Comment:
I see the issue with Proxy, however, you don't need to recreate this, vue exposes a `toRaw` helper that extracts the target (original value) of the Proxy.

```suggestion
     toRaw(this.authClient).login({
```

Reply:
Excellent, this works and is exactly what I needed!